tests/test/coherence.rs

```diff
@@ -264,9 +264,9 @@ fn fundamental_traits() {
     lowering_error! {
         program {
             #[upstream] trait Sized { }
-            #[upstream] struct str { }
+            #[upstream] struct str2 { } // str is already known to be !Sized, so make a new str type
```

Comment:
This test was broken after adding `str` (and removing the struct declaration here), and I couldn't find a suitable `!Sized` replacement in std that would fit the explanation. Or, maybe I'm misinterpreting the reason the test failed (can't apply `#[upstream]` to `str` because it's predefined?).

Reply:
This is fine (probably, this should have never been called `str`. Can you instead make this be named `Foo`?

Reply:
The reason for using `str` is noted in the comment above the test:
> We want to enable negative reasoning about some traits. For example, consider the str type.
We know that str is never going to be Sized and we have made a decision to allow people to
depend on that. The following two impls are rejected as overlapping despite the fact that we
know that str will never be Sized.

I can remove the explanation around `str`, since it no longer fits with `Foo`.

Reply:
Yeah, this is just a coherence check. Nothing special about `Sized`/`str`. You're changes are good :)